frontend/routes/Wifi.jsx

```diff
@@ -10,44 +28,268 @@ export class Component extends React.Component {
+  };
+
   return (
-    <table>
-      <thead>
-        <tr>
-          {Object.keys(data[0]).map(s => (
-            <th key={s}>{s}</th>
-          ))}
-        </tr>
-      </thead>
-      <tbody>
-        {data.map((row, i) => (
-          <tr key={i}>
-            {Object.values(row).map((s, i) => (
-              <td key={i}>{s}</td>
-            ))}
-          </tr>
+    <TableHead>
+      <TableRow>
+        {headCells.map((headCell) => (
+          <TableCell
+            key={headCell.id}
+            align='center'
+            padding={headCell.disablePadding ? 'none' : 'default'}
+            sortDirection={orderBy === headCell.id ? order : false}
+          >
+            <TableSortLabel
+              active={orderBy === headCell.id}
+              direction={orderBy === headCell.id ? order : 'asc'}
+              onClick={createSortHandler(headCell.id)}
+            >
+              {headCell.label}
+              {orderBy === headCell.id ? (
+                <span className={classes.visuallyHidden}>
+                  {order === 'desc' ? 'sorted descending' : 'sorted ascending'}
+                </span>
+              ) : null}
+            </TableSortLabel>
+          </TableCell>
         ))}
-      </tbody>
-    </table>
+      </TableRow>
+    </TableHead>
   );
 }
+
+EnhancedTableHead.propTypes = {
+  classes: PropTypes.object.isRequired,
+  onRequestSort: PropTypes.func.isRequired,
+  order: PropTypes.oneOf(['asc', 'desc']).isRequired,
+  orderBy: PropTypes.string.isRequired,
+};
+
+const useToolbarStyles = makeStyles((theme) => ({
+  root: {
+    padding: theme.spacing(2),
+  },
+  highlight:
+    theme.palette.type === 'light'
+      ? {
+          color: theme.palette.secondary.main,
+          backgroundColor: lighten(theme.palette.secondary.light, 0.85),
+        }
+      : {
+          color: theme.palette.text.primary,
+          backgroundColor: theme.palette.secondary.dark,
+        },
+  title: {
+    flex: '1 1 100%',
+  },
+}));
+
+const EnhancedTableToolbar = (props) => {
+  const classes = useToolbarStyles();
+  return (
+    <Toolbar>
+        <Typography className={classes.title} variant="h6" id="tableTitle" component="div">
+          Wifi 
+        </Typography>
+
+    </Toolbar>
+  );
+};
+
+EnhancedTableToolbar.propTypes = {
+};
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: '90%',
+    margin: theme.spacing(2),
+  },
+  paper: {
+    width: '100%',
+    padding: theme.spacing(2),
+    marginBottom: theme.spacing(1),
+    
+  },
+  table: {
+    minWidth: 550,
+  },
+  visuallyHidden: {
+    border: 0,
+    clip: 'rect(0 0 0 0)',
+    height: 1,
+    margin: -1,
+    overflow: 'hidden',
+    padding: 0,
+    position: 'absolute',
+    top: 20,
+    width: 1,
+    
+  },
+}));
+
+function descendingComparator(a, b, orderBy) {
+  let is_num = false;
+  headCells.map((element)=>{
+    if(element.id == orderBy)
+      is_num = is_num || element.numeric
+  });
```

Comment:
Why not make `headCells` a dictionary with `id` as the key?

This way you can avoid this search.

Reply:
Noted